### DISCUSSION TITLE
Cant get bed level with bltouch ender 3 v2

### DISCUSSION THREAD

**Discussion by baldwin4200** [Jyers/Marlin#1655] [Q&A]
So I've been trying to get my bltouch to work for a couple weeks now i have tried jyers 3x3 5x5 and UBL 10x10 and now im on UBL 15x15 version 2.0.1 and i still cant get good first layers over all parts of my bed. No matter what i do it is way to close in the middle of the bed and too far away on corners. is it possible my bed is too warped for a abl to work? when i manually leveled it i had the same issue i could only print in the middle of my bed since it would always be too far away on the edges for good first layers. im making sure to manually level it the best i can before i make a new mesh. here is my start gcode im not too sure if its the issue or what's going on. 

; Ender 3 Custom Start G-code
G92 EO ; Reset Extruder 
G28 ; Home all
G29 A ;
G29 L0 ;
G29 J2 ;	
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X0.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line
G1 X0.4 Y200.0 Z0.3 F5000.0 ; Move to side a little
G1 X0.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish

**Comment by esseivan** [comment]
Have you properly set the Z-Offset ? That is the offset between the bl-touch probe and the hotend.
If you just upgraded to 2.0.1, it may most surely have been reset to the default value.

**Reply by baldwin4200** [reply]
Yes i have set the z offset but i have to bring it pretty low i have it set at -2.92 right now and its way too close in the middle of the bed it wont even extrude but the corners its so far away its almost printing in mid air.

**Reply by esseivan** [reply]
I don't know if you're very familiar with leveling. I may be wrong, but it looks like you're not doing the right way for leveling.
You should set the z offset at the middle, which should be home (home all) by default. And I would suggest the bltouch version either 3x3 or 5x5.

Basically the leveling process is : 
1. Create a new mesh (level) 
2. Calibrate the z offset while the probe is in the middle. It should automatically home when enabling the live adjustment setting (I don't remember the exact name). You should juat be able to slide a sheet of paper without friction, you can always sligthly adjust later if the print isn't good. 

Peronally, I stopped bothering about manually adjusting the bed with the knobs. I just put them in a position where the spring is under pressure, not loose nor maximum compressed.
Also I try to have my bed more or less flat (but not leveled) by having two opposites corners being max and min values, and the two others somewhere in the middle.

**Reply by baldwin4200** [reply]
That is exactly how i am setting my z offset. i will try the 5x5 again and get back to you and see if i have any better luck with it. but i think i will have the same issue again my bed is super warped in the middle. even when i first got it months ago i could never get it leveled manually I've always been limited to printing in the middle of the bed. is it possible my bed is too warped for the bltouch to compensate for it?

**Reply by Sirnut** [reply]
What is your probe offset?

**Reply by baldwin4200** [reply]
-44 -7
